mesonbuild/interpreterbase.py

```diff
@@ -178,18 +200,23 @@ def wrapped(*wrapped_args, **wrapped_kwargs):
 class FeatureCheckBase:
     "Base class for feature version checks"
 
-    def __init__(self, feature_name, version):
-        self.feature_name = feature_name
-        self.feature_version = version
+    # Class variable, shared across all instances
+    #
+    # Format: {subproject: {feature_version: set(feature_names)}}
+    feature_registry = {}  # type: Dict[str, Dict[str, Set[str]]]
```

Comment:
Using `Typing.ClassVar[]` around the annotation to denote this.

Reply:
Also, fixed